### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Migrate Browser Switch Testing and Release to GitHub Actions

 ### Checklist

 ~- [ ] Added a changelog entry~

- @sshropshire

### CODE REVIEW

**Review by jaxdesmarais — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -16,113 +16,91 @@ dependencies {
...
-            // the user canceled and returned to your app
-            // return uri is null
-            break;
-        case ERROR:
-            // there was an error browser switching
-            // Some possible issues you may encounter are:
-            // - There are no activities installed that can handle a URL.
-            // - The integration is not setup correctly.
-            break;
+In the above example, notice the encoded `callbackURL` parameter is forwarded to the website that will be loaded. The callback url should have the same custom scheme set in `BrowserSwitchOptions`. When this URL is loaded by the site, the Android OS will re-direct the user to the deep link destination `Activity` defined in the `AndroidManifest.xml`.
```
[the remark this answers is not in the document]
Comment (sshropshire) [reply README.md:74]: Yeah I guess we could say "must" in stead of "should" to make it clear that the url scheme needs to match whatever's specified in the `<intent-filter>`. Other than that it's tough, that parameter could be called anything. It all depends on the website they're forwarding the callback url to.

File: README.md [hunk 2]
```diff
@@ -16,113 +16,89 @@ dependencies {
 }
 ```
 
-Then, declare `BrowserSwitchActivity` in your `AndroidManifest.xml`:
+To preview the latest work in progress builds, add the following SNAPSHOT dependency in your `build.gradle`:
+
+```groovy
+dependencies {
+  implementation 'com.braintreepayments.api:browser-switch:2.3.3-SNAPSHOT'
```
Comment (scannillo) [inline README.md:23]: NIT - maybe let's not tie this to a version to future-proof it.

Comment (scannillo) [reply README.md:23]: Nevermind, just saw the `task changeREADMEVersion` below! 👏

Comment (sshropshire) [reply README.md:23]: This actually gets updated by the GitHub actions script to be the most recent.

**Comment by scannillo** [comment]
Can we remove the `Rakefile` in this PR too?

**Comment by sshropshire** [comment]
> Can we remove the `Rakefile` in this PR too?

Yeah I've been thinking a bash script feels like the best bet. We can DRY up the GitHub Actions template a bit and also run some CI tasks locally.

**Review by jcnoriega — APPROVED** [review]

**Pull request merged**